ceph_iscsi_config/utils.py

```diff
@@ -28,14 +28,27 @@ class CephiSCSIInval(CephiSCSIError):
     pass
 
 
-def shellcommand(command_string):
+def parse_disk_meta(disk):
+    pool = None
+    datapool = None
+    image = None
+    pools, image = disk.split('/')
+    try:
+        pool, datapool = pools.split('+')
+    except ValueError:
+        pool = pools
+        pass
+    return pool, datapool, image
```

Comment:
I commented on the README, but just to reiterate, `parse_disk_meta` shouldn't be needed.

Reply:
No, even we remove the `rep+ec` style this still will be needed in other places. Such as when loading and parsing the configures from the `rbd` pool.

Reply:
Can you give a specific example?  Like I said, I think `rep+ec` should come up only when pretty-printing.  It shouldn't leak outside of that and therefore should never be parsed.

Reply:
Such as from the logs:

```
187 2022-01-04 20:35:46,876    DEBUG [common.py:120:_read_config_object()] - _read_config_object reading the config object
188 2022-01-04 20:35:46,876    DEBUG [common.py:170:_get_ceph_config()] - (_get_rbd_config) config object contains 'b'{\n    "created": "2021/12/21 02:    18:46",\n    "discovery_auth": {\n        "mutual_password": "",\n        "mutual_password_encryption_enabled": false,\n        "mutual_username":     "",\n        "password": "",\n        "password_encryption_enabled": false,\n        "username": ""\n    },\n    "disks": {\n        "rbd+ecpool/bl    ockX1": {\n            "allocating_host": "node01",\n            "backstore": "user:rbd",\n            "backstore_object_name": "rbd.blockX1",\n                "controls": {},\n            "created": "2021/12/21 02:39:38",\n            "datapool": "ecpool",\n            "image": "blockX1",\n                "owner": "node02",\n            "pool": "rbd",\n            "pool_id": 2,\n            "updated": "2021/12/21 02:39:49",\n            "wwn":     "e121dbf4-10e2-4f60-83ca-4480da5875ce"\n        },\n        "rbd/blockX0": {\n            "allocating_host": "node01",\n            "backstore": "u    ser:rbd",\n            "backstore_object_name": "rbd.blockX0",\n            "controls": {},\n            "created": "2021/12/21 02:19:21",\n                "datapool": null,\n            "image": "blockX0",\n            "owner": "node01",\n            "pool": "rbd",\n            "pool_id": 2,\n                "updated": "2021/12/21 02:20:03",\n            "wwn": "a0f1b2c6-ad93-4761-aa47-91ea9034a442"\n        }\n    },\n    "epoch": 11,\n    "ga    teways": {\n        "node01": {\n            "active_luns": 1,\n            "created": "2021/12/21 02:19:39",\n            "updated": "2021/12/21 0    2:20:03"\n        },\n        "node02": {\n            "active_luns": 1,\n            "created": "2021/12/21 02:19:44",\n            "updated": "20    21/12/21 02:39:49"\n        }\n    },\n    "targets": {\n        "iqn.2003-01.com.redhat.iscsi-gw:ceph-igw0": {\n            "acl_enabled": true,\n                "auth": {\n                "mutual_password": "",\n                "mutual_password_encryption_enabled": false,\n                "mutua    l_username": "",\n                "password": "",\n                "password_encryption_enabled": false,\n                "username": ""\n                },\n            "clients": {\n                "iqn.1994-05.com.redhat:client": {\n                    "auth": {\n                        "mutual    _password": "",\n                        "mutual_password_encryption_enabled": false,\n                        "mutual_username": "",\n                            "password": "",\n                        "password_encryption_enabled": false,\n                        "username": ""\n                        },\n                    "group_name": "",\n                    "luns": {\n                        "rbd+ecpool/blockX1": {\n                                "lun_id": 1\n                        },\n                        "rbd/blockX0": {\n                            "lun_id": 0\n                            }\n                    }\n                }\n            },\n            "controls": {},\n            "created": "2021/12/21 02:19:33",\n                "disks": {\n                "rbd+ecpool/blockX1": {\n                    "lun_id": 1\n                },\n                "rbd/blockX0"    : {\n                    "lun_id": 0\n                }\n            },\n            "groups": {},\n            "ip_list": [\n                "172.    16.219.128",\n                "172.16.219.138"\n            ],\n            "portals": {\n                "node01": {\n                    "gateway    _ip_list": [\n                        "172.16.219.128",\n                        "172.16.219.138"\n                    ],\n                    "ina    ctive_portal_ips": [\n                        "172.16.219.138"\n                    ],\n                    "portal_ip_addresses": [\n                            "172.16.219.128"\n                    ],\n                    "tpgs": 2\n                },\n                "node02": {\n                        "gateway_ip_list": [\n                        "172.16.219.128",\n                        "172.16.219.138"\n                    ],\n                        "inactive_portal_ips": [\n                        "172.16.219.128"\n                    ],\n                    "portal_ip_addresses": [\    n                        "172.16.219.138"\n                    ],\n                    "tpgs": 2\n                }\n            },\n            "u    pdated": "2021/12/21 02:40:03"\n        }\n    },\n    "updated": "2021/12/21 02:40:03",\n    "version": 11\n}''
```

We can see:

`"disks": { "rep+ecpool/blockX1": { "lun_id": 1 }, ....`

We will save the disks as `rep+ec/image` style.

Reply:
Yeah, and also we can save the `datapool` by using one `datapool: ecpool` par in the config, for current code I think I need to change a lot or recode many places and won't benefit much from it.

Reply:
No, we should all disks in `pool/image` style, or, in other words, by the image spec.  For the CLI, the image spec has the following format: `pool/image@snap`.  Here in ceph-iscsi, I guess it is `pool/image/snap` (i.e. `@` isn't used) but this minor difference is irrelevant.

The `pool` in `pool/image@snap` is *always* the metadata pool.  The (optional) data pool is never part of the spec.  Let's not deviate from the rest of RBD in ceph-iscsi.

Reply:
Hmmm, okay, let me recode it this week.